### PR TITLE
Initiate rpc module

### PR DESCRIPTION
**Pull request by posgnu**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,7 +15,15 @@
...
     readme = f.read()
 
-requires = ["rlp", "coincurve", "dataclasses; python_version < '3.7'", "pycrypto", "jsondatabase"]
+requires = [
+    "rlp",
+    "coincurve",
+    "dataclasses; python_version < '3.7'",
+    "pycrypto",
+    "jsondatabase",
+    "json-rpc",
```
Comment (foriequal0) [inline setup.py]: It is a library to make a server. I think you've followed their client library, but it doesn't contain any dependencies from their package. It only requests to the server through `requests`, the HTTP request library, and dump/load with built-in `json` library'
https://github.com/pavlov99/json-rpc/blob/master/examples/client.py

I've tried to use `jsonrpcclient[requests]` before: https://jsonrpcclient.readthedocs.io/en/latest/api.html

File: codechain/rpc/base.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import json
+from typing import Callable
+from typing import Union
+
+import requests
+
+
+class rpc_base:
```
Comment (foriequal0) [inline codechain/rpc/base.py]: You should be consistent with the class name. We have three different naming convention on the class name. `snake_case` on `rpc_base`, and `camelCase` on `keystoreManager`, and `PascalCase` on the others. Also, we have inconsistency in file names.

Comment (foriequal0) [inline codechain/rpc/base.py]: If you decide to use `jsonrpcclient`, then this class would be some useless class. It supports automatic id generation, json deserialization, error handlings.

**Review by foriequal0 — CHANGES_REQUESTED** [review]

**Comment by foriequal0** [comment]
I understand that you've tried to implement the RPC classes by inheriting the base class. It was how the original codechain-rpc is implemented. However I think it is generally not a good design though. It is bad for the reusing the code, and the testing. You can reuse the feature without inheriting a base class by injecting the feature. Think about what we did in this PR: https://github.com/CodeChain-io/codechain-sdk-python/pull/48/files#diff-e774b762519803eb930f012626c79f09R13

File: codechain/rpc/request.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class RpcRequest:
+    def __init__(self, node: str):
+        self.client = HTTPClient(node)
+
+    def call(self, request):
+        response = self.client.send(request)
+
+        if not response.ok:
```
Comment (foriequal0) [inline codechain/rpc/request.py]: I think they already do this: https://github.com/bcb/jsonrpcclient/blob/master/jsonrpcclient/client.py#L176
Then this `RpcRequest` class isn't necessary, I think.

**Review by foriequal0 — CHANGES_REQUESTED** [review]

File: codechain/rpc/request.py (hunk 3 above)
Comment (posgnu) [reply codechain/rpc/request.py]: I removed it.

**Review by foriequal0 — APPROVED** [review]

**Pull request merged**